assets/src/domain/eventEditor/services/apollo/queries/tickets/useFetchTickets.ts

```diff
@@ -17,37 +16,35 @@ const useFetchTickets = (skipFetch: boolean = null): FetchEntitiesResult<Tickets
 	const skip = skipFetch !== null ? skipFetch : !datetimeIn.length || isLoaded(TypeName.tickets);
 
 	const toaster = useToaster();
-	const toasterMessage = 'initializing tickets';
+	const loadingToastKey = useRef(toaster.generateKey());
 
 	const { data, error, loading } = useQuery<TicketsList>(query, {
 		...options,
 		skip,
 		onCompleted: (): void => {
 			setIsLoaded(TypeName.tickets, true);
-			toaster.dismiss(toasterMessage);
+			toaster.dismiss(loadingToastKey.current);
 			toaster.success({ message: `tickets initialized` });
 		},
 		onError: (error): void => {
 			setIsError(TypeName.tickets, true);
-			toaster.dismiss(toasterMessage);
+			toaster.dismiss(loadingToastKey.current);
 			toaster.error({ message: error });
 		},
 	});
 
 	useEffect(() => {
 		setIsLoading(TypeName.tickets, loading);
-	}, [loading]);
-
-	if (!initialized) {
-		toaster.loading(loading, toasterMessage);
-		toaster.error({ message: error });
-	}
 
-	useEffect(() => {
-		if (!initialized && !loading) {
-			setInitialized(true);
+		if (loading && !toaster.isToastOpen(loadingToastKey.current)) {
+			toaster.loading({
+				key: loadingToastKey.current,
+				message: 'initializing tickets',
+			});
+		} else if (!loading) {
+			toaster.dismiss(loadingToastKey.current);
```

Comment:
Can this be somehow extracted, I've seen similar code being repeated above.

Reply:
Yes, sure

Reply:
I concur!

Reply:
Done

Reply:
and pushed?